maintenance/pre-commit.sh

```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env bash
+
+set -Eeuo pipefail
+
+YUNODIR="$(realpath "$(dirname "$0")"/..)"
+SHFMT="$YUNODIR"/maintenance/shfmt.sh
+
+help() {
+    echo "pre-commit.sh ACTION [LANG]"
+    echo "  Run common pre-commit checks to make CI happy"
+    echo "  ACTIONS:"
+    echo "    - check: provide recommendations"
+    echo "    - fix: apply recommendations automatically"
+    echo "  LANG: python|bash, by default applies to both languages"
+}
+
+check_deps() {
+    for dep in "$@"; do
+        if ! command -v "$dep" >/dev/null 2>&1; then
+            echo "Missing dependency: $dep. Please install it first."
+            exit 1
+        fi
+    done
+}
+
+check_bash() {
+    check_deps shfmt
+    "$SHFMT" -d "$YUNODIR"/helpers/helpers
+    "$SHFMT" -d "$YUNODIR"/helpers/helpers.v2.1.d/*
+    "$SHFMT" -d "$YUNODIR"/helpers/helpers.v2.d/*
+    "$SHFMT" -d "$YUNODIR"/hooks/*
```

Comment:
I see one major flaw at calling multiple times shfmt: if you use `check`, and you have issues in multiple helpers dirs, it will fail on the first and NOT show issues from the subsequent ones.

So i suggest calling it **once** with all the files on the command line.